src/common/LRUSet.h

```diff
@@ -0,0 +1,145 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <functional>
+#include <boost/intrusive/list.hpp>
+#include <boost/intrusive/unordered_set.hpp>
+#include "include/encoding.h"
+
+/// Combination of an LRU with fast hash-based membership lookup
+template<class T, int NUM_BUCKETS=128>
```

Comment:
it's a little different than simple_cache because there is no payload, but yeah, these could probably be collapsed... maybe by making the template take ``void`` or similar.  the key result being that simple_cache should use intrusive, which should cut down allocations by like 3x.